# A robust predictor for hERG channel blockade based on deep learning meta-feature ensembles data prepration and representation

In [1]:
import pandas as pd
import numpy as np
df = pd.read_csv("../data/train_validation_cardio_tox_data.csv")
print("Total Row: ",df.shape[0])
print("Total Column: ", df.shape[1])
df.head()

/var/folders/x6/r97phsxs5qb8n7m2m39tbx0m0000gn/T/ipykernel_72781/1553959414.py:3: DtypeWarning: Columns (118,119,120,124,125,126,130,131,132,136,137,138,142,143,144,148,149,150,154,155,156,160,161,162,166,167,168,172,173,174,268,269,270,274,275,276,280,281,282,286,287,288,292,293,294,298,299,300,304,305,306,310,311,312,316,317,318,322,323,324,327,328,329,332,333,334,337,338,339,342,343,344,347,348,349,352,353,354,357,358,359,362,363,364,367,368,369,372,373,374,377,378,379,382,383,384,387,388,389,392,393,394,397,398,399,402,403,404,407,408,409,412,413,414,417,418,419,422,423,424,480,481,482,483,493,494,495,496,720) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv("../data/train_validation_cardio_tox_data.csv")


Total Row:  12620
Total Column:  997


ACTIVITY                                             smiles        ABC  \
0         1                  Fc1ccc(-n2cc(NCCN3CCCCC3)nn2)cc1F  17.108462   
1         0  COc1cc(N2Cc3ccc(Sc4ccc(F)cc4)nc3C2=O)ccc1OCCN1...  26.886637   
2         0  CCOC(=O)[C@H]1CC[C@@H](N2CC(NC(=O)CNc3nn(C(N)=...  28.187229   
3         0  N[C@@H](Cn1c(=O)cnc2ccc(F)cc21)C1CCC(NCc2ccc3c...  27.105416   
4         0  O=C(NC1COc2cccc(-c3ccnc(CO)c3)c2C1)c1ccc(OCC(F...  25.880229   

       ABCGG  nAcid  nBase  nAromAtom  nAromBond  nAtom  nHeavyAtom  ...  \
0  13.312773      0      1         11         11     41          22  ...   
1  19.283801      0      1         18         18     60          34  ...   
2  20.640811      0      1          9         10     65          36  ...   
3  18.346730      0      2         16         17     61          34  ...   
4  18.463716      0      0         18         18     53          33  ...   

      SRW09      SRW10     TSRW10          MW       AMW  WPath  WPol  Zagreb1  \
0  6.580639   9.781828  69.261944  307.160852  7.491728   1258    29      112   
1  7.412764  10.412922  86.178228  479.167891  7.986132   4035    52      182   
2  7.050123  10.607303  87.262633  510.220238  7.849542   4957    56      192   
3  0.000000  10.500041  70.568018  466.212867  7.642834   4122    56      184   
4  0.000000  10.391239  69.196044  459.140591  8.663030   3658    51      174   

   Zagreb2  mZagreb2  
0      128  4.888889  
1      215  7.472222  
2      226  7.750000  
3      217  7.305556  
4      201  7.152778  

[5 rows x 997 columns]

In [2]:
from rdkit import Chem
from mordred import Calculator, descriptors
calc = Calculator(descriptors, ignore_3D = True)
mol = Chem.MolFromSmiles("CCOC(=O)C1(CCN(C)CC1)c1ccccc1")
x=calc(mol)
print(len(x))

RDKit WARNING: [20:46:43] Enabling RDKit 2019.09.3 jupyter extensions
[20:46:43] Enabling RDKit 2019.09.3 jupyter extensions


1613


In [3]:
df1 = pd.read_csv("../data/external_test_set_pos.csv")
total_activity = df1.loc[:,'ACTIVITY']
total_smiles = df1.loc[:,'smiles']
mols = [Chem.MolFromSmiles(smi) for smi in total_smiles]
# did not work as dataframe on multiple mols
    # df2 = calc.pandas(mols)
    # print("Total Row: ",df2.shape[0])
    # print("Total Column: ", df2.shape[1])
    # df2.head()
from multiprocessing import freeze_support
freeze_support()
calc = Calculator(descriptors)
print(calc.pandas(mols))

RemoteError: 
---------------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/santuchal/opt/anaconda3/lib/python3.8/site-packages/numpy/core/__init__.py", line 22, in <module>
    from . import multiarray
  File "/Users/santuchal/opt/anaconda3/lib/python3.8/site-packages/numpy/core/multiarray.py", line 12, in <module>
    from . import overrides
  File "/Users/santuchal/opt/anaconda3/lib/python3.8/site-packages/numpy/core/overrides.py", line 7, in <module>
    from numpy.core._multiarray_umath import (
ImportError: dlopen(/Users/santuchal/opt/anaconda3/lib/python3.8/site-packages/numpy/core/_multiarray_umath.cpython-38-darwin.so, 2): Symbol not found: ___addtf3
  Referenced from: /Users/santuchal/opt/anaconda3/lib/python3.8/site-packages/numpy/core/../.dylibs/libquadmath.0.dylib
  Expected in: /usr/lib/libSystem.B.dylib
 in /Users/santuchal/opt/anaconda3/lib/python3.8/site-packages/numpy/core/../.dylibs/libquadmath.0.dylib

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/santuchal/opt/anaconda3/lib/python3.8/multiprocessing/managers.py", line 202, in handle_request
    request = c.recv()
  File "/Users/santuchal/opt/anaconda3/lib/python3.8/multiprocessing/connection.py", line 251, in recv
    return _ForkingPickler.loads(buf.getbuffer())
  File "/Users/santuchal/opt/anaconda3/lib/python3.8/site-packages/mordred/__init__.py", line 3, in <module>
    from ._base import (
  File "/Users/santuchal/opt/anaconda3/lib/python3.8/site-packages/mordred/_base/__init__.py", line 3, in <module>
    from .util import is_missing
  File "/Users/santuchal/opt/anaconda3/lib/python3.8/site-packages/mordred/_base/util.py", line 1, in <module>
    from ..error import MissingValueBase
  File "/Users/santuchal/opt/anaconda3/lib/python3.8/site-packages/mordred/error/__init__.py", line 6, in <module>
    import numpy as np
  File "/Users/santuchal/opt/anaconda3/lib/python3.8/site-packages/numpy/__init__.py", line 150, in <module>
    from . import core
  File "/Users/santuchal/opt/anaconda3/lib/python3.8/site-packages/numpy/core/__init__.py", line 48, in <module>
    raise ImportError(msg)
ImportError: 

IMPORTANT: PLEASE READ THIS FOR ADVICE ON HOW TO SOLVE THIS ISSUE!

Importing the numpy C-extensions failed. This error can happen for
many reasons, often due to issues with your setup or how NumPy was
installed.

We have compiled some common reasons and troubleshooting tips at:

    https://numpy.org/devdocs/user/troubleshooting-importerror.html

Please note and check the following:

  * The Python version is: Python3.8 from "/Users/santuchal/opt/anaconda3/bin/python"
  * The NumPy version is: "1.21.0"

and make sure that they are the versions you expect.
Please carefully study the documentation linked above for further help.

Original error was: dlopen(/Users/santuchal/opt/anaconda3/lib/python3.8/site-packages/numpy/core/_multiarray_umath.cpython-38-darwin.so, 2): Symbol not found: ___addtf3
  Referenced from: /Users/santuchal/opt/anaconda3/lib/python3.8/site-packages/numpy/core/../.dylibs/libquadmath.0.dylib
  Expected in: /usr/lib/libSystem.B.dylib
 in /Users/santuchal/opt/anaconda3/lib/python3.8/site-packages/numpy/core/../.dylibs/libquadmath.0.dylib

---------------------------------------------------------------------------

## As per the code 
* 1. In attached train_validation_cardio_tox_data.csv file there are 995 descriptor where as in new descriptor there are 1613 descriptor. May be If I got time I'll add later and generate all descriptor with new version of mordred 
* 2. Seems that the mordred package has changed a little bit and drop multiple mols support for pandas. 